crates/bevy_core_pipeline/src/auto_exposure/pipeline.rs

```diff
@@ -0,0 +1,162 @@
+use super::compensation_curve::{
+    AutoExposureCompensationCurve, AutoExposureCompensationCurveUniform,
+};
+use bevy_asset::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_render::{render_resource::*, renderer::RenderDevice, texture::Image, view::ViewUniform};
+use std::num::NonZeroU64;
+
+#[derive(Resource)]
+pub struct AutoExposurePipeline {
+    pub histogram_layout: BindGroupLayout,
+    pub histogram_shader: Handle<Shader>,
+}
+
+#[derive(Component)]
+pub struct ViewAutoExposurePipeline {
+    pub histogram_pipeline: CachedComputePipelineId,
+    pub mean_luminance_pipeline: CachedComputePipelineId,
+    pub state: Buffer,
+    pub compensation_curve: Handle<AutoExposureCompensationCurve>,
+    pub uniform: AutoExposureUniform,
+    pub metering_mask: Handle<Image>,
+}
+
+#[derive(ShaderType, Clone, Copy)]
+pub struct AutoExposureUniform {
+    pub(super) min_log_lum: f32,
+    pub(super) inv_log_lum_range: f32,
+    pub(super) log_lum_range: f32,
+    pub(super) low_percent: f32,
+    pub(super) high_percent: f32,
+    pub(super) speed_up: f32,
+    pub(super) speed_down: f32,
+    pub(super) exp_up: f32,
+    pub(super) exp_down: f32,
+}
+
+#[derive(PartialEq, Eq, Hash, Clone)]
+pub enum Pass {
+    Histogram,
+    Average,
+}
+
+pub const METERING_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(12987620402995522466);
+
+pub const HISTOGRAM_BIN_COUNT: u64 = 64;
+
+impl FromWorld for AutoExposurePipeline {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+
+        Self {
+            histogram_layout:
+                render_device.create_bind_group_layout(
+                    "compute histogram bind group",
+                    &[
```

Comment:
Could possibly use https://github.com/bevyengine/bevy/pull/10224 ?

Reply:
Ahh, missed that one. That would clean things up a lot indeed.